### PR TITLE
Give the offboard wire's command conversion one owner

### LINKED ISSUE #597: Give the offboard wire's command conversion one owner
Two modules convert between the offboard policy wire and `roboarm.command`, and neither owns the boundary:

| Site | Shape it handles |
|---|---|
| `utils/serialization.py` — `_pack` / `_unpack` | the `__cmd__` envelope, via `to_wire` / `from_wire` |
| `policy/remote.py` — `RemoteSession._as_command` | a command served as a bare `{'type': ..., ...}` mapping |

Both answer the same question — a served command becoming the typed object the drivers dispatch on — and splitting them costs twice.

**A generic utility carries domain knowledge.** `utils/serialization.py` opens by saying so itself:

> This module currently knows about `roboarm.command` directly. If we accumulate more domain types that need wire treatment (gripper commands, observation packets, etc.), replace the inline dispatch with a generic registry / `__to_wire__` protocol so utils stays domain-agnostic.

**Nothing answers "where does the wire become a command?"** The conversion has since sat in `utils/serialization.py`, then `policy/codec.py`, then `policy/remote.py`. Six separate review findings on #576 flagged comments in four modules, each trying to state this boundary's contract from wherever its author was standing — `codec.py`, `remote.py`, `keys.py`, and the tests. Each finding was right that its site neither states nor enforces the rule. The rule still has no site that does.

## The shape that already exists next door

`simulator/env_server/protocol.py` owns the env-server boundary's wire codec in one module, deliberately positronic-free so a benchmark's interpreter can import it. The offboard boundary has no equivalent: `offboard/` holds `client.py` and `server.py`, and its encoding lives in `utils/`.

## Scope

- One owner for the offboard wire's command conversion, accepting both shapes it already accepts
- `utils/serialization.py` drops its `roboarm` import
- `offboard/client.py` and `offboard/server.py` are the users

Both shapes stay supported. The bare mapping is what a partner endpoint sends today; #576 carries tests that pin it through a live socket, and those must keep passing.

## Out of scope, and why

- **The env-server wire (#592).** A different vocabulary on purpose: `cartesian` against `cartesian_pos`, `joint_vel`/`dq` against `joint_delta`/`velocities`, `hold` for `None`, no `reset`, and `_wire_command` raises on a `CartesianDelta` outside the env control frame rather than encoding it. Its far end must not import positronic, so it cannot share this code.
- **`dataset/serializers.py::Serializers.robot_command`.** A storage schema, unfolding a command into `.pose` / `.joints` / `.pose_delta` columns. Datasets outlive protocols.

## What has to be checked before starting

Moving command handling out of `utils/serialization.py` is safe only if the remote **dataset** wire (`dataset/remote.py`, `dataset/remote_server/server.py`) never carries a `CommandType`. `Serializers.robot_command` unfolds commands into arrays at record time, so a stored episode should hold vectors rather than command objects — inferred from the serializer, not verified end to end. This is load-bearing: the two wires share `serialise`/`deserialise`.

## Separate finding, not part of this scope

`policy/recording.py::_command_field_arrays` calls `roboarm_command.to_wire` to name the columns it writes, so recorded column names derive from the network encoding. Renaming a wire field would silently change what a recording is keyed on, and datasets written either side of that change would stop matching. Worth its own issue if it survives a closer look.

### PR DESCRIPTION
**Pull request by vertix**
## Summary

Two modules converted between the offboard policy wire and `roboarm.command`, and neither owned the boundary: the `__cmd__` envelope in `utils/serialization.py`, and the bare `{'type': ..., ...}` mapping in `RemoteSession._as_command`. Both answer the same question, and the conversion had already migrated through three homes without landing in one.

`positronic/offboard/protocol.py` now owns it, mirroring how `simulator/env_server/protocol.py` owns the env-server boundary's codec:

- `serialise` / `deserialise` — the shared msgpack encoding plus the `__cmd__` envelope
- `typed_commands` — the bare `to_wire` mapping a partner endpoint sends, which nothing but the channel it sits at tells from any other dict

`offboard/client.py` and `offboard/server.py` are its users. `InferenceSession.infer` hands up typed commands, so `RemoteSession` no longer knows commands exist and the conversion has one site instead of two. Both wire shapes stay supported.

`utils/serialization.py` drops its `roboarm` import and publishes `pack` / `unpack` as the extension point a boundary chains from its own msgpack hooks — the domain-agnostic form its own TODO asked for.

**Two things worth a reviewer's attention:**

**The `RobotStatus` envelope is deleted, not moved.** Nothing produces one on this wire: `Serializers.robot_state` strips the status before an observation is built, and `Harness._build_obs` is the only observation producer. It was added in #381 for the `robot_state.error` field that #474 deleted, and no test ever exercised it. Carrying it into a module documenting what this wire carries would have stated something false.

**The issue's load-bearing precondition is verified.** The remote *dataset* wire shares `serialise`/`deserialise`, so moving command handling out is only safe if it never carries a `CommandType`. It cannot: episode statics are hard-validated to JSON scalars (`episode.py:34`) and persisted as JSON, and `Kind` has only `NUMERIC` and `IMAGE` — there is no object-valued signal.

Out of scope, per the issue: the env-server wire (#592), `dataset/serializers.py::Serializers.robot_command`, and the `policy/recording.py::_command_field_arrays` finding.

## Test plan

- Full suite: 1117 passed, 9 skipped
- The mocked decode tests move to `test_offboard.py` as tests of the new owner — mocking `InferenceSession` now bypasses the conversion entirely, so they would have pinned nothing where they were
- The live-socket round trip stays in `test_remote_policy.py` and was watched failing with the decode removed, confirming it pins the conversion rather than passing incidentally
- A new case asserts the two wire shapes converge on the same typed command through a real msgpack round trip
- `ruff check` / `ruff format` clean; `check-rules` clean across all 13 rules (2 honoured `hardcoded-keys` waivers, both the wire-spelling pattern #576 established)

Closes #597

### CODE REVIEW

File: positronic/offboard/protocol.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import numpy as np
+
+from positronic import keys
+from positronic.drivers.roboarm import command
+from positronic.utils import serialization
+
+
+def _pack(obj):
+    if isinstance(obj, command.CommandType):
+        return {b'__cmd__': command.to_wire(obj)}
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/offboard/protocol.py]: Agreed — fixed in 2d47199c.

`_CMD = b'__cmd__'` is now defined once in `offboard/protocol.py`, above the two hooks that write and read it.

Extended it to the same shape next door: `utils/serialization.py` spelled `__ndarray__`, `__npgeneric__` and `__jpeg__` twice each, and the `frames`/`ndim`/`data`/`dtype`/`shape` fields once in the writer and once in the reader. Those are functions this PR already touched, and leaving them literal beside protocol.py's constant would have made the constant look arbitrary — so they get a constants block too.

File: positronic/offboard/client.py [hunk 2]
```diff
@@ -87,7 +84,7 @@ def infer(self, obs: dict[str, Any]) -> Any:
         if isinstance(response, dict) and 'error' in response:
             raise RuntimeError(f'Server error: {response["error"]}')
 
-        return response['result']
+        return typed_commands(response['result'])
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/offboard/client.py]: Agreed — fixed in 2d47199c.

`RESULT` now lives in `offboard/protocol.py`, which both `client.py` and `server.py` already import.

Doing only `result` would have left a half-measure: `error`, `status`, `meta` and `message` sit right beside it with the identical cross-module duplication (server writes, client reads). So the whole frame vocabulary moved — `STATUS`, `MESSAGE`, `META`, `RESULT`, `ERROR`.

One thing went further than constants. The four `status` *values* are a closed set the handshake dispatches on with an if-ladder, which `primitive-type` says is the case where the set is closed and an enum only states it. They are now a `Status(StrEnum)`, converted once as the frame arrives. That let `_handshake` drop its `(waiting, loading)` tuple and its unknown-status fallthrough — an unknown value fails at the conversion instead. `StrEnum` serialises as the same plain string, so the wire is byte-identical, and each error path was checked to produce the message it produced before:

| frame | message |
|---|---|
| `{'error': 'boom'}` | `Server error: boom` |
| `{'status': 'error', 'error': 'boom'}` | `Server error: boom` |
| `{'status': 'error'}` | `Server error: Unknown error` |
| `{'status': 'bogus'}` | `Unexpected server response: {'status': 'bogus'}` |
| `{}` | `Unexpected server response: {}` |

File: positronic/offboard/protocol.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import msgpack
+import numpy as np
+
+from positronic import keys
+from positronic.drivers.roboarm import command
+from positronic.utils import serialization
+
+# The frame envelope every message travels in: ``STATUS`` frames until the server reports itself ready and
+# hands over its ``META``, then one ``RESULT`` or ``ERROR`` per inference. The server writes these and the
+# client reads them, so both import them from here.
```
[the remark this answers is not in the document]
Comment (vertix) [reply positronic/offboard/protocol.py]: Agreed — fixed in f7ceefb7, which landed before this round arrived.

The sentence you flagged was exactly the caller narration:

> The server writes these and the client reads them, so both import them from here.

It answered "why does this module exist?" rather than stating anything that must hold of the constants, so it is gone. What remains is the wire-envelope invariant alone — `STATUS` frames until the server reports ready and hands over its `META`, then one `RESULT` or `ERROR` per inference — which is the frame ordering a reader cannot get from five string assignments.

The same pass cut three more comments failing the same test: the module docstring's paragraph on the env-server boundary next door (explains a neighbouring module and defends a design choice — that belongs in the commit message), the `_CMD` comment (restated the ten lines beneath it), and the second sentence of `_typed`'s docstring (restated the `isinstance` guard on the following line).

One comment was kept deliberately: the note on why `_typed` runs the values through `np.asarray` before `from_wire`. That is a constraint the code cannot state — the wire carries sequences where `from_wire` expects arrays.

`protocol.py` is 85 lines down to 76.
